src/ucp/api/ucp.h

```diff
@@ -1089,7 +1089,9 @@ typedef struct ucp_worker_attr {
     size_t                max_am_header;
 
     /**
-     * Cumulative bandwidth this worker is able to provide
+     * Cumulative bandwidth this worker is able to provide.
+     * If @ref UCP_WORKER_ADDRESS_FLAG_NET_ONLY is in address_flags, only
```

Comment:
what is the use case for this?  If the application uses CS connection scheme and does not need worker address, but also wants to get bandwidth between two nodes - it will get full bandwidth (including SHM devices)

Reply:
Yeah, it doesn't make sense to sum up bandwidth of all devices
need to return max bandwidth of a single device
adding @abellina here - what would be the preferred API for Spark to retrieve the UCX expected bandwidth? If we do it per endpoint, it might be too late; since we might not want to create the endpoints in the first place

Reply:
Agree, the sum is not going to help in this case. 

The way I thought about this was "I connected to endpoint X, what BW is available". So If endpoint X is in the same machine, that could be NVLink, it could be PCIe, etc. To me getting info about "what an endpoint can achieve" is more important than what the worker has available, in isolation.

For this PR, if I understand it correctly, it's all about what my single worker can do. In this case, we don't know what transports are enabled for a peer, but it is still useful because we can ask the worker what it has available. It is like running `ucx_info -d` with a different api, and so at the very least a Spark executor can tell you "I can do cuda_ipc, and RoCE" and that seems like a good tool overall, and could lead to us falling back to the built-in shuffle if no accelerated transports are possible.

Reply:
maybe worker attr should contain both - intra node and inter node BW?

Reply:
So, shall we return max BW instead of summing? @yosefe  @abellina